### PR TITLE
Actions, ability to bypass proxy on request, get_raw method

### PR DESCRIPTION
**Pull request by dovreshef**
I have added a couple of things:

1. Option to disable going through proxy, even when exist. 
This helps in case when you want for example to update Zabbix installation on localhost, and you don't want the script to go through the proxy.
2. get_raw method. Useful in case you want to expand selection for interfaces on hosts, for example. Since the usual method only enables you to specify key values for the filter properties.
3. The get_id method will call directly to the api, not through get_full_data, and will only asks for the id and nothing else. It helps performance when checking a large number of templates, for example. Also helps debugging, when scrolling through the output :).
4. Actions implementation.

### CODE REVIEW

File: lib/zabbixapi/basic/basic_logic.rb [hunk 1]
```diff
@@ -82,16 +91,24 @@ def all
...
+          :filter => {
+            indentify.to_sym => data[indentify.to_sym]
+          },
+          :output => [key]
+        }
+      )
       id = nil
-      result.each { |item| id = item[key.to_sym].to_i if item[indentify.to_sym] == data[indentify.to_sym] }
+      result.each { |item| id = item[key].to_i if item[indentify] == data[indentify] }
       id
```
Comment (evtuhovich) [inline lib/zabbixapi/basic/basic_logic.rb]: @dovreshef i'm not sure about this change, i need some time to check.

**Comment by evtuhovich** [comment]
@dovreshef thanx for you PR, but now i have to errors in specs

```

Failures:

  1) application when name not exists get_id should return nil
     Failure/Error: expect(zbx.items.get_id(:host => @item)).to be_kind_of(NilClass)
       expected 23691 to be a kind of NilClass
     # ./spec/item.rb:41:in `block (4 levels) in <top (required)>'

  2) application when name exists delete should delete item from zabbix
     Failure/Error: expect(zbx.items.get_id(:id => @itemid)).to be_nil
       expected: nil
            got: 23693
     # ./spec/item.rb:126:in `block (4 levels) in <top (required)>'

Finished in 6.15 seconds
92 examples, 2 failures

Failed examples:

rspec ./spec/item.rb:40 # application when name not exists get_id should return nil
rspec ./spec/item.rb:125 # application when name exists delete should delete item from zabbix
```

**Comment by dovreshef** [comment]
@evtuhovich thanks for the comments.

I've fixed the tests, and added tests for actions.

**Comment by evtuhovich** [comment]
Sorry, but you break this thing

```ruby
z.hosts.get_id('host' => 'host_18005114')
```

I'm sure a lot of people use string keys with zabbixapi, so i don't want to break this.

**Comment by dovreshef** [comment]
@evtuhovich No problem. I'll fix it and add a test. 
Is that ok? 
Any other error you have seen?

**Comment by evtuhovich** [comment]
@dovreshef  Wait for a 15 minutes, i've cherry-pick some of your commits

**Comment by evtuhovich** [comment]
@dovreshef i've cherry-picked some of your commits, so merge master to your branch.

Yes, only this problem with string keys.

**Comment by dovreshef** [comment]
Ok. I'll fix and create a new PR with the difference.

**Pull request closed without merging**